src/main/java/com/internet/shop/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package com.internet.shop.service.impl;
+
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.dao.UserDao;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.model.User;
+import com.internet.shop.service.UserService;
+import java.util.List;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public User create(User user) {
+        userDao.create(user);
+        shoppingCartDao.create(new ShoppingCart(user.getId()));
```

Comment:
According to comments to my HW we shouldn't do it (for example for admin users we don't need cart)

Reply:
Agree, didn't think about admin users. Refactored.